gateway/src/shard/processor/socket_forwarder.rs

```diff
@@ -41,18 +37,12 @@ impl SocketForwarder {
         #[cfg(feature = "tracing")]
         tracing::debug!("starting driving loop");
 
-        loop {
-            let timeout = sleep(Self::TIMEOUT).fuse();
-            tokio::pin!(timeout);
-
-            let rx = Box::pin(self.rx.recv().fuse());
-            let tx = Box::pin(self.stream.next().fuse());
+        let timeout = sleep(Self::TIMEOUT);
+        tokio::pin!(timeout);
 
-            let select_message = future::select(rx, tx);
-
-            match future::select(select_message, timeout).await {
-                // `rx` future finished first.
-                Either::Left((Either::Left((maybe_msg, _)), _)) => {
+        loop {
+            tokio::select! {
```

Comment:
the dsl of select! is not obvious. code should be obvious to read. if this needs to be improved, improve it in a way that doesn't use the macro